.github/workflows/release.yml

```diff
@@ -0,0 +1,29 @@
+name: Release to Maven Central
+
+on:
+  workflow_dispatch:
+  release:
+
+jobs:
+  release:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v4
+
+        - name: Set up JDK 17
+          uses: actions/setup-java@v4
+          with:
+            java-version: '17'
+            distribution: 'temurin'
+
+        - name: Setup Gradle
+          uses: gradle/actions/setup-gradle@v3
+
+        - name: Release
+          env:
+            AUTOMATED_MAVEN_RELEASE_PGP_SECRET: ${{ secrets.AUTOMATED_MAVEN_RELEASE_PGP_SECRET }}
+            AUTOMATED_MAVEN_RELEASE_GITHUB_APP_PRIVATE_KEY: ${{ secrets.AUTOMATED_MAVEN_RELEASE_GITHUB_APP_PRIVATE_KEY }}
+            AUTOMATED_MAVEN_RELEASE_SONATYPE_PASSWORD: ${{ secrets.AUTOMATED_MAVEN_RELEASE_SONATYPE_PASSWORD }}
```

Comment:
Just to note that this is absolutely normal practice across most artifact publishing ecosystems, whether it's NPM, Maven, PyPI or whatever, but it _does_ mean that theses secret are loaded into environment variables that pretty much any dependency in the build or tests can access, which is a much broader exposure than's actually _required_!

This is what [`gha-scala-library-release-workflow`](https://github.com/guardian/gha-scala-library-release-workflow) specifically avoids by isolating different build phases in their own containers - see the [notes](https://github.com/guardian/gha-scala-library-release-workflow/blob/main/docs/security-design.md) on this aspect of the security design.

Reply:
This is partly what I wanted to talk to you about. I think the way I've set it up at the moment will work as is.

Having looked through the scala release workflow, I believe we should be able to use the `Sonatype Release` step with little change - Gradle can create the artifacts locally, and the release step can upload them securely independently.

The step I'm not sure about is signing. At the moment, the build+sign is a single step that Gradle executes. I think we may be able to separate them into separate steps though I haven't looked at how to do, or how much effort it is. What I don't know is whether the GPG signing for scala artifacts is the same as what Gradle expects (likely/hopefully), or is it very different?

My intention is to first get this working with the default Gradle workflow as here.
I'm also starting to document a "how to" which we can use when we start publishing Coverdrop and Bridget too.
Then we can work to separate the signing & release steps and maybe create a variant of the Scala flow for Gradle based projects.